### PR TITLE
Completed API endpoint stretch goals

### PR DESCRIPTION
**Pull request by jamesquillen**

### CODE REVIEW

File: spec/requests/reptiles_spec.rb [hunk 1]
```diff
@@ -40,32 +40,59 @@
...
+  reptile = Reptile.first
+
+
+      new_reptile_params = {
+      reptile: {
+        name: 'Shakespeare',
+        age: 7,
+        enjoys: 'Eating crickets, a lot. Being pet on his little head. Adventuring',
+        image: 'https://images.unsplash.com/photo-1619153832248-6458a856d91f?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1931&q=80'
+      }
```
Comment (SunkissedQueen) [inline spec/requests/reptiles_spec.rb:64]: Please make sure that nested items are indented.

File: app/controllers/reptiles_controller.rb [hunk 2]
```diff
@@ -21,6 +21,9 @@ def update
     end
 
     def destroy
+        reptile = Reptile.find(params[:id])
+        reptile.destroy
+        render json: reptile
```
Comment (SunkissedQueen) [inline app/controllers/reptiles_controller.rb]: Are you sure you want to render the data entry that was just removed? Is there some other way that you can show that the entry was deleted?

**Review by SunkissedQueen — COMMENTED** [review]
Congratulations on tackling the stretch goals! I left some notes to consider when rendering json on the destroy endpoint. Please look over indentation.

File: spec/requests/reptiles_spec.rb (hunk 1 above)
Comment (jamesquillen) [reply spec/requests/reptiles_spec.rb:64]: We adjusted some indentation.

File: app/controllers/reptiles_controller.rb (hunk 2 above)
Comment (jamesquillen) [reply app/controllers/reptiles_controller.rb]: We changed it so it renders Reptile.all instead.

**Review by SunkissedQueen — COMMENTED** [review]
Will you be handling the it blocks that are not tested in a describe block on the API validation card?

**Comment by mosscantrell** [comment]
Yes, those should all be fixed on the validations branch!

**Review by SunkissedQueen — APPROVED** [review]
Approving because changes to the test structure are being handled in the api validations PR

**Pull request merged**